src/test/java/test_user_registration/TestUserExistsInteractor.java

```diff
@@ -0,0 +1,128 @@
+package test_user_registration;
+
+import data_access.Database;
+import entities.user_entities.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import use_cases.user_registration_use_cases.ISendVerificationCode;
+import interface_adapters.user_registration_interface_adapters.UserExistsPresenter;
+import interface_adapters.user_registration_interface_adapters.UserExistsOutputView;
+import use_cases.user_registration_use_cases.createMailMan;
+
+public class TestUserExistsInteractor {
+    //Objects used for testing
+    // The below anonymous database is created to test cases where the user does not exist.
+    Database userDontExist = new Database() {
```

Comment:
I like how you are creating simplified anonymous classes for the purpose of testing